themes/default/content/blog/api-gateway-to-eventbridge/index.md

```diff
@@ -0,0 +1,371 @@
+---
+title: "API Gateway to EventBridge with Pulumi"
+date: 2022-01-07
+draft: true
+meta_desc: These are words that add up to at least 50 characters.
+meta_image: meta.png
+authors:
+    - christian-nunciato
+tags:
+    - aws
+    - api-gateway
+    - eventbridge
+    - lambda
+---
+
+If you're familiar with [Amazon API Gateway](https://aws.amazon.com/api-gateway/), you know it's all about making it easier to provision and manage a web API. Maybe you've used it, as I have, with [Crosswalk]({{< relref "/docs/guides/crosswalk/aws" >}}), our AWS extension library, to stand up a REST API and handle requests with AWS Lambda functions:
+
+```typescript
+import * as awsx from "@pulumi/awsx";
+
+// Create a new API Gateway instance.
+const api = new awsx.apigateway.API("my-api", {
+    routes: [
+        {
+            // Define an HTTP endpoint.
+            method: "GET",
+            path: "/things",
+
+            // Handle requests with an AWS Lambda function.
+            eventHandler: async (apiGatewayEvent) => {
+                return {
+                    statusCode: 200,
+                    body: JSON.stringify([
+                        "thingOne",
+                        "thingTwo",
+                    ]),
+                };
+            },
+        },
+    ],
+});
+
+// Export the API's public URL. 🎉
+export const apiUrl = api.url;
+```
+
+I love this abstraction, and I use it all the time; it's an incredibly convenient way to solve a really common problem. But if you look closely at the code, you'll see that it's making a fairly strong assumption about how you'll be handling HTTP requests --- namely, that you'll do so with a single Lambda function, and that that function will always return a JavaScript object [of a particular shape](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-integration-settings-integration-response.html).
+
+Indeed, this arrangement represents the API contract of a [Lambda _proxy_ integration](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html) --- API Gateway integrations come in [many shapes and sizes](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations.html); Lambda integrations just happen to be one of the more popular --- and much of the time, an approach like this one works out just fine. But depending on the needs of the application, it might not always be the best fit.
+
+Imagine you were building a print-on-demand service, for example, and you wanted to expose an API to let your customers upload  documents and have them converted into orders. On AWS, you might reach for API Gateway, as above, to define an HTTP method and route (`POST /uploads`, say), wire it up to an AWS Lambda, and have the Lambda parse the upload, write the order to a database, and return a response. Visually, such a design might look something like this:
+
+![Figure 1: Write new orders to the database](./figure-1.png)
```

Comment:
Good suggestion; I didn't realize these were so verbose! I'll revisit 'em, thanks for pointing it out.

Reply:
I'm by no means an expert on alt text; I just know it should be descriptive enough to convey the meaning of the image. I generally get decently descriptive on flowcharts like this because a lot of context is lost without it.

Reply:
heres a guide on writing alt text that id recommend: https://webaim.org/techniques/alttext/
specifically, id recommend removing the "The figure shows" language. Does the same information already appear in text form outside of the alt text? If so, it could potentially be reduced.

Reply:
I've updated the alt text for each of these diagrams to express what they (focusing on content and context--thanks for sharing that doc @susanev).